### DISCUSSION TITLE
Binding on @Query

### DISCUSSION THREAD

**Discussion by volnei** [micronaut-projects/micronaut-data#2434] [General]
Micronaut appers to binding only first field

```java
	@Query(value = "SELECT DATE(e.created_at) as date, e.event AS event, COUNT(e.id) AS total FROM events e WHERE app = :app GROUP BY e.event, DATE(e.created_at)", nativeQuery = true, readOnly = true)
	public abstract Mono<List<EventStatsDTO>> stats(String app);
```

DTO
```java
import java.time.LocalDate;

import io.micronaut.core.annotation.Introspected;
import lombok.Data;

@Data
@Introspected
public class EventStatsDTO {
    
    private String event;
    private LocalDate date;
    private Long total;
}
```

Results in 
```
["2023-08-04"]
```

**Comment by radovanradic** [comment]
Hi, I tried to recreate sample app for this and it looks ok (it's just jdbc repository, but that shouldn't make difference).
[mn-mysql-dto.zip](https://github.com/micronaut-projects/micronaut-data/files/12262053/mn-mysql-dto.zip)
Can you take a look and update what's diff so we can try to reproduce.

**Comment by volnei** [comment]
@radovanradic thank you for your reply and attention!

I fixed it by using `Flux<EventStatsDTO>` insted of  `Mono<List<EventStatsDTO>>` but I don't know why the second one doest work
